### PR TITLE
implement menu-entry load file

### PR DESCRIPTION
**Pull request by awgrover**
and some install notes.

I put comments in the "files changed" view/tab.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,30 @@
 # YADA (Yet Another Dataflow Attempt)
 
- <img width="300" alt="Screen Shot 2022-05-31 at 9 39 05 PM" src="https://user-images.githubusercontent.com/27078897/171679946-9c6dbce3-871a-41c7-8ec7-2b8d6825be00.gif">
+<img width="300" alt="Screen Shot 2022-05-31 at 9 39 05 PM" src="https://user-images.githubusercontent.com/27078897/171679946-9c6dbce3-871a-41c7-8ec7-2b8d6825be00.gif">
 
-Some design considerations for YADA
+## Install
```
Comment (awgrover) [inline README.md:5]: Because I had to figure it out.

File: download.js [hunk 2]
```diff
@@ -7,3 +7,12 @@ export function download(filename, text) {
...
 }
+
+export function readFile(file) {
+  // file is an event.target.files[0], from change event of a <input type="file")
+  return new Promise((resolve, reject) => {
+    let fr = new FileReader();
+    fr.onload = x=> resolve(fr.result);
+    fr.readAsText(file);
+})}
+
```
Comment (awgrover) [inline download.js:18]: used in index.js

File: index.js [hunk 3]
```diff
@@ -89,12 +88,15 @@ const ACTIONS = {
...
-    dispatch("RENDER");
+  LOAD_FILE(file, state) {
+    readFile(file).then(json => {
+      const { nodes, connections, name } = decode(json);
+      state.nodes = nodes;
+      state.connections = connections;
+      state.name = name;
+      dispatch("RENDER");
+      dispatch("SET_NAME", file);
+    });
```
Comment (awgrover) [inline index.js:99]: full implementation for use by event handler.

Note the SET_NAME. I'm setting the name to the file, which seems what a user would expect. However, there is a `name` in the .json...

File: nodes/async_adder.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ export default {
 
     const result = new Promise(resolve => {
       setTimeout(() => {
-        resolve(x+y+30);
+        resolve(x+y);
```
Comment (awgrover) [inline nodes/async_adder.js:25]: over zealous?

File: view.js [hunk 5]
```diff
@@ -188,6 +188,10 @@ export default function view(state) {
           spellcheck="false"
           @blur=${e => dispatch("SET_NAME", { name: e.target.innerText })}>${state.name}</div>
         <div class="menu-item" @click=${() => dispatch("SAVE_TO_FILE")}>save</div>
+        <div class="menu-item" @click=${() => document.getElementById("load_file").click()}>load</div>
+        <!-- hidden -->
+        <input type="file" accept=".json" style="display:none;" id="load_file" name="file" @input=${e => dispatch("LOAD_FILE", e.target.files[0] )}>
+
```
Comment (awgrover) [inline view.js:194]: Using the hidden <input> pattern.

File: README.md (hunk 1 above)
Comment (leomcelroy) [reply README.md:5]: Great to add. I added Vite to the project to make this simpler for some people and to reserve the option to install packages if we must (though sometimes I feel it's advantageous to make it more difficult).

**Pull request merged**